packages/anthropic/src/convert-to-anthropic-messages-prompt.ts

```diff
@@ -251,12 +251,18 @@ export function convertToAnthropicMessagesPrompt({
               }
 
               case 'reasoning': {
-                anthropicContent.push({
-                  type: 'thinking',
-                  thinking: part.text,
-                  signature: part.signature!,
-                  cache_control: cacheControl,
-                });
+                // omit reasoning content if no signature is present. middleware
+                // could have extracted it from an older model's text content
+                // with no associated signature, and anthropic API
+                // requires signature presence even for older models.
+                if (part.signature) {
+                  anthropicContent.push({
+                    type: 'thinking',
+                    thinking: part.text,
+                    signature: part.signature,
+                    cache_control: cacheControl,
+                  });
+                }
```

Comment:
We might want to add warnings, since we are omitting information some users might wonder about.

Reply:
Done.